apps/extension/src/entrypoints/selection.content/selection-toolbar/speak-button.tsx

```diff
@@ -0,0 +1,109 @@
+import { Icon } from '@iconify/react'
+import { useAtomValue, useSetAtom } from 'jotai'
+import { useCallback, useState } from 'react'
+import { toast } from 'sonner'
+import { configFieldsAtomMap } from '@/utils/atoms/config'
+import { getProviderApiKey, getProviderBaseURL } from '@/utils/config/helpers'
+import { isTooltipVisibleAtom, selectionContentAtom } from './atom'
+
+export function SpeakButton() {
+  const [isSpeaking, setIsSpeaking] = useState(false)
+  const selectionContent = useAtomValue(selectionContentAtom)
+  const setIsTooltipVisible = useSetAtom(isTooltipVisibleAtom)
+  const providersConfig = useAtomValue(configFieldsAtomMap.providersConfig)
+
+  const handleClick = useCallback(async () => {
+    if (!selectionContent) {
+      toast.error('No text selected')
+      return
+    }
+
+    // Check if OpenAI is configured
+    const openaiProvider = providersConfig.find(p => p.provider === 'openai' && p.enabled)
+    if (!openaiProvider) {
+      toast.error('OpenAI provider is not configured or enabled')
+      return
+    }
+
+    const apiKey = getProviderApiKey(providersConfig, openaiProvider.id)
+    if (!apiKey) {
+      toast.error('OpenAI API key is not configured')
+      return
+    }
+
+    setIsSpeaking(true)
+    setIsTooltipVisible(false)
+
+    try {
+      const baseURL = getProviderBaseURL(providersConfig, openaiProvider.id) || 'https://api.openai.com/v1'
+
+      const response = await fetch(`${baseURL}/audio/speech`, {
+        method: 'POST',
+        headers: {
+          'Authorization': `Bearer ${apiKey}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          model: 'tts-1',
+          input: selectionContent,
+          voice: 'alloy',
+          speed: 1.0,
+        }),
+      })
+
+      if (!response.ok) {
+        const errorData = await response.json().catch(() => ({}))
+        throw new Error(errorData.error?.message || `HTTP error! status: ${response.status}`)
+      }
+
+      const audioBlob = await response.blob()
+      const audioUrl = URL.createObjectURL(audioBlob)
+      const audio = new Audio(audioUrl)
+
+      audio.onended = () => {
+        URL.revokeObjectURL(audioUrl)
+        setIsSpeaking(false)
+      }
+
+      audio.onerror = () => {
+        URL.revokeObjectURL(audioUrl)
```

Comment:
You can use React Query to reduce the amount of code and simplify state management.